### PR TITLE
[MLMC] Add XMC

### PR DESCRIPTION
**Pull request by riccardotosi**
Dear all,

This PR adds the XMC library to Kratos.

XMC is a library designed for Hierarchical Monte Carlo algorithms, and is integrated with Kratos. Additionally, the library is designed for running in distributed environments.

The library is added as external library inside the Multilevel Monte Carlo application, and the Multilevel Monte Carlo application is added to the continuous integration. Tests for checking the integration between XMC and Kratos have been added. Examples of usage have been added.

The update of the library will be performed manually, and applied directly to `MultilevelMonteCarloApplication/external_libraries/XMC`.

Since the library, as well as the MultilevelMonteCarlo Application, requires the [PyCOMPSs library](https://github.com/KratosMultiphysics/Kratos/wiki/How-to-run-multiple-cases-using-PyCOMPSs) to run, empty definitions are added in `MultilevelMonteCarloApplication/external_libraries/PyCOMPSs`.

At compilation time, the user can decide if he prefers using the mockup PyCOMPSs library, or one manually installed by him/her. As well, the use can decide if he prefers using the XMC library version inside Kratos, or use a different installation.

I kindly ask to @KratosMultiphysics/technical-committee to review this PR.

### CODE REVIEW

File: .github/workflows/ci.yml
Comment (philbucher) [inline .github/workflows/ci.yml]: this should not be changed
I think you could put this in an `__init__.py`

**Review by philbucher — COMMENTED** [review]
just browsing through
Please reduce the size of the mdpa you are adding (should be less than 1000 lines)

**Comment by philbucher** [comment]
just to say that if this goes in it should be `squashed` otherwise we have another large file in the git-history

**Comment by philbucher** [comment]
as I wrote before probably the right place to add things to the path is in the `__init__.py` of the application

**Comment by roigcarlo** [comment]
Yep, at this point is just testing why the hell the module is not found on the CI. (Seems to work everywhere else...)

**Review by philbucher — CHANGES_REQUESTED** [review]
Blocking because I want to give this a final review before it is being merged

**Comment by philbucher** [comment]
guys there are still some huge mdpas in this PR
also some *bin files

Please clean this

**Comment by riccardotosi** [comment]
You are right @philbucher , my bad. Thanks (but it was not ready for reviewing)

**Comment by roigcarlo** [comment]
@philbucher Maybe we can try the Merge-Squash here so we don't have to care about blobs in this PR. what do you say?

**Comment by philbucher** [comment]
> @philbucher Maybe we can try the Merge-Squash here so we don't have to care about blobs in this PR. what do you say?

[quoting myself](https://github.com/KratosMultiphysics/Kratos/pull/6729#issuecomment-621772468), yes this is necessary here IMO, otherwise we again have some more mdpas in the history :)

**Comment by philbucher** [comment]
To me looks good
Some questions:
- Should this app be in the CI?
    - What are the long-term plans?
    - Who are the official maintainers?
- Since there were many big files involved this should be merged with a squash-commit

**Comment by riccardotosi** [comment]
> Should this app be in the CI? 

I don't think I am the one who should answer this question. Are there some criteria? If so, I can try to answer them.

> Who are the official maintainers?

Official maintainers are @RiccardoRossi and myself. Regarding the XMC library, official maintainers are Riccardo Rossi, myself, a team at EPFL and one at BSC.

> What are the long-term plans?

`MultilevelMonteCarloApplication` follows the start of the ExaQUte application, as well as the XMC library. I cannot speak for what will happen after the project, probably @RiccardoRossi has a wider view on this question.

> Since there were many big files involved this should be merged with a squash-commit.

Ok.

**Comment by RiccardoRossi** [comment]
> To me looks good
> Some questions:
> 
>     * Should this app be in the CI?
>       
>       * What are the long-term plans?
>       * Who are the official maintainers?
> 
>     * Since there were many big files involved this should be merged with a squash-commit

Hi,

i would like this to be in the CL if possible. This app is interesting longer term, apart for the ExaQUte project.

i am completely ok with the squash.

also @riccardotosi @marcnunezc this is your creature, so you deserve being the official mantainers

**Comment by philbucher** [comment]
green light :vertical_traffic_light: from me then

@roigcarlo @rubenzorrilla @pooyan-dadvand ?

**Comment by roigcarlo** [comment]
Ok to me also :+1:

**Review by roigcarlo — DISMISSED** [review]
Re-Approving changes for the new XMC branch

**Comment by riccardotosi** [comment]
From my side, I am ready to merge (with squash commit).

**Review by philbucher — APPROVED** [review]

**Comment by riccardotosi** [comment]
Seems merged correctly with the squash. Thanks @roigcarlo @philbucher for the help!

**Pull request merged**
